### ISSUE TITLE
Metadata not including nested type when a list of the type is prop of req DTO

### ISSUE THREAD

**Issue by georgehemmings** [ServiceStack/Issues#483]
I've found a problem with the metadata Op page when using a nested type. For example the Address type  from the sample below is not included on the Op page. If I move the Address class outside of the Hello class it does appear on the Op page. Also if I just had `public Address OneAddress { get; set; }` instead of a list it does appear.

```csharp
[Route("/hello/{Name}")]
public class Hello : IReturn<HelloResponse>
{
    public string Name { get; set; }
    public List<Address> Addresses { get; set; }

    public class Address
    {
        public string HouseNumber { get; set; }
        public string Postcode { get; set; }
    }
}
```

Sample proj: https://github.com/georgehemmings/SSMetadataWrongType

**Comment by mythz** [comment]
Thanks, should now be fixed in this commit: https://github.com/ServiceStack/ServiceStack/commit/8d247c1028e1c8b7adaeb2915ac49186aa049323

This change is available from v4.5.1 that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet).

**Comment by georgehemmings** [comment]
Thanks
